test/sagemaker_tests/pytorch/training/conftest.py

```diff
@@ -141,10 +144,11 @@ def pytest_configure(config):
 
 
 def pytest_runtest_setup(item):
-    if item.config.getoption("--efa"):
-        efa_tests = [mark for mark in item.iter_markers(name="efa")]
-        if not efa_tests:
-            pytest.skip("Skipping non-efa tests")
+    efa_tests = [mark for mark in item.iter_markers(name="efa")]
+    if item.config.getoption("--efa") and not efa_tests:
+        pytest.skip("Skipping non-efa tests due to --efa flag")
+    elif not item.config.getoption("--efa") and efa_tests:
+        pytest.skip("Skipping efa tests because --efa flag is missing")
```

Comment:
Is it worth logging here in an `else` statement?

Reply:
This is two separate boolean variables that need to be true in "xnor".